### PR TITLE
Update to use cached functions

### PR DESCRIPTION
**Pull request by jasonbahl**
As part of a VIP GO code review, we were asked to update our codebase to use cached functions. We updated our controlled code, and are working to submit PRs to codebases we don't directly control

### CODE REVIEW

File: co-authors-plus.php [hunk 1]
```diff
@@ -784,8 +792,9 @@ function coauthors_update_post( $post_id, $post ) {
 	}
 
 	function has_author_terms( $post_id ) {
-		$terms = wp_get_object_terms( $post_id, $this->coauthor_taxonomy, array( 'fields' => 'ids' ) );
-		return ! empty( $terms ) && ! is_wp_error( $terms );
+		$term_objects = get_the_terms( $post_id, $this->coauthor_taxonomy );
+		$terms = ( ! is_wp_error( $term_objects ) ) ? wp_list_pluck( $term_objects, 'term_id' ) : array();
```
Comment (mjangda) [inline co-authors-plus.php]: The list pluck is unnecessary here. We can just do `return ! is_wp_error( $term_objects ) && ! empty( $term_objects )`

**Comment by mjangda** [comment]
I think we should drop the uses of `wpcom_vip_get_term_by` (i.e. VIP-specific functions) here (the `id` lookups use `get_term` anyway, which should be cached already?)

**Comment by bcampeau** [comment]
For the instances where we're using `id` as the first param, we don't even really need `get_term_by` at all. `get_term` would be sufficient there and that's definitely a cached function.

For the other usage by slug, I'm fine with the check for the existence of `wpcom_vip_get_term_by` since there really isn't another way to make it a cached function.

**Comment by jasonbahl** [comment]
Going through some old open PRs of mine. . .is this worth merging with some changes or should I close this?

**Comment by rebeccahum** [comment]
Closing as `wpcom_vip_get_term_by()` is deprecated now, see: https://vip.wordpress.com/functions/wpcom_vip_get_term_by/

**Pull request closed without merging**